Persephone/Controllers/QueueViewController.swift

```diff
@@ -0,0 +1,183 @@
+//
+//  QueueController.swift
+//  Persephone
+//
+//  Created by Daniel Barber on 2019/2/01.
+//  Copyright © 2019 Dan Barber. All rights reserved.
+//
+
+import Cocoa
+import mpdclient
+
+class QueueViewController: NSViewController, NSOutlineViewDataSource, NSOutlineViewDelegate {
+  var queue: [MPDClient.Song] = []
+  var queuePos: Int32 = -1
+
+  var queueIcon: NSImage? = nil
+
+  let systemFontRegular = NSFont.systemFont(ofSize: 13, weight: .regular)
+  let systemFontBold = NSFont.systemFont(ofSize: 13, weight: .bold)
+
+  struct SongItem {
+    var song: MPDClient.Song
+    var queuePos: Int
+  }
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+
+    queueView.columnAutoresizingStyle = .uniformColumnAutoresizingStyle
+
+    NotificationCenter.default.addObserver(
+      self,
+      selector: #selector(stateChanged(_:)),
+      name: MPDClient.stateChanged,
+      object: AppDelegate.mpdClient
+    )
+
+    NotificationCenter.default.addObserver(
+      self,
+      selector: #selector(queueChanged(_:)),
+      name: MPDClient.queueChanged,
+      object: AppDelegate.mpdClient
+    )
+
+    NotificationCenter.default.addObserver(
+      self,
+      selector: #selector(queuePosChanged(_:)),
+      name: MPDClient.queuePosChanged,
+      object: AppDelegate.mpdClient
+    )
+  }
+
+  @objc func stateChanged(_ notification: Notification) {
+    guard let state = notification.userInfo?[MPDClient.stateKey] as? mpd_state
+      else { return }
+
+    setQueueIcon(state)
+  }
+
+  @objc func queueChanged(_ notification: Notification) {
+    guard let queue = notification.userInfo?[MPDClient.queueKey] as? [MPDClient.Song]
+      else { return }
+
+    self.queue = queue
+
+    queueView.reloadData()
+  }
+
+  @objc func queuePosChanged(_ notification: Notification) {
+    guard let queuePos = notification.userInfo?[MPDClient.queuePosKey] as? Int32
+      else { return }
+
+    let oldSongRowPos = Int(self.queuePos + 1)
+    let newSongRowPos = Int(queuePos + 1)
+    self.queuePos = queuePos
+
+    setQueuePos(oldSongRowPos: oldSongRowPos, newSongRowPos: newSongRowPos)
+
+    queueView.reloadData(
+      forRowIndexes: [oldSongRowPos, newSongRowPos],
+      columnIndexes: [0, 1]
+    )
+  }
+
+  func setQueueIcon(_ state: mpd_state) {
+    switch state {
+    case MPD_STATE_PLAY:
+      self.queueIcon = NSImage(named: NSImage.Name("playButton"))
```

Comment:
Similar to some of @louis-antonopoulos’s other suggestions, this image name constants could be extracted as extensions on `NSImage.Name`, and/or you could define a convenience extension on `NSImage` with a series of `static var`s that gives access to your bundled images.